XmlFile.js

```diff
@@ -128,60 +133,20 @@ function isPrimitive(type) {
     return ["boolean", "number", "integer", "string"].indexOf(type) > -1;
 }
 
-/**
- * Gets type of array based on provided schema.
- *
- * TODO: Add support for "anyOf" and "oneOf" type definitions.
- */
-function getArrayTypeFromSchema(schema, root) {
-    if (schema.type !== "array") {
-        return null;
-    }
-
-    if (typeof(schema.items["$ref"]) !== 'undefined') {
-        // substitute the referenced schema for this one
-        var refname = schema.items["$ref"];
-        var otherschema = root["$$refs"][refname];
-        if (!otherschema) {
-            console.log("Unknown reference " + refname + " while parsing " +
-                    this.pathName + " with schema " + root["$id"]);
-            return;
-        }
-        schema = otherschema;
-    } else {
-        schema = schema.items;
-    }
-
-    var allowedTypes = ["string", "integer", "number", "boolean", "object"];
-
-    if (schema.type && allowedTypes.indexOf(schema.type) > -1) {
-        return schema.type;
-    }
-
-    // Default type is string for compatibility reasons.
-    return 'string';
-}
-
-/**
- * Converts each element of one-dimensional array to a primitive type.
- */
-function convertArrayElementsToType(array, type) {
-    if (!ilib.isArray(array)){
-        return array;
-    }
-
-    return array.map(function (item) {
-        return convertValueToType(item, type);
-    });
-}
-
 /**
  * Converts value to a primitive type.
  */
 function convertValueToType(value, type) {
     switch (type) {
         case "boolean":
-            return value === "true";
+            switch (value) {
+                case "false":
+                case "no":
+                case "0":
```

Comment:
Should we consider empty string as `false` too?

Reply:
For XML files, if you have an attribute that has no value, that attribute is considered to equal `true`:

`<item name="foo" checked>`

Reply:
Having said that, your comment highlighted a problem on line 336 where values that are falsy do not get processed...